docs/diagrams/ListActivityDiagram.puml

```diff
@@ -2,23 +2,20 @@
 
 start
 :user enters list command;
-if (has parameter?) then (yes)
-    :identify parameter
-    is of what type;
-    if (is patient?) then (yes)
-        :update list to show
-              all patients;
-    elseif (is donor?) then (yes)
-        :update list to show
-              all donors;
-    elseif (is doctor?) then (yes)
-        :update list to show
-              all doctors;
-    else (invalid type)
-        :display error message;
-    endif
-else (no)
-    :list all persons;
+:identify parameter of list command;
+if (no parameter?) then (yes)
```

Comment:
I'm not very familiar with UML so I'm not going to suggest uml changes as of now. 

I will approach you in person regarding how to draw this out, it's quite difficult to explain via text.

Reply:
noted